### ISSUE TITLE
Preprocessing SlimPajama

### ISSUE THREAD

**Issue by amy-hyunji** [microsoft/Samba#26]
Hi! Thank you for releasing a great work!
I am currently working on replicating the pretraining code for the 410M model and have encountered a few questions regarding the data processing in Slimpajama (`scripts/prepare_slimpajama.py`) and dataloader. 

1. In the code, the chunk_size is set to 2049 * 1024 (https://github.com/microsoft/Samba/blob/3da69d13348105092424cf870990d27cad545727/scripts/prepare_slimpajama.py#L83). This value seems quite large, and as a result, no files are being saved during processing (https://github.com/microsoft/Samba/blob/3da69d13348105092424cf870990d27cad545727/lit_gpt/packed_dataset.py#L113). Did you use a different value for chunk_size (other than 2049 * 1024) when processing Slimpajama? Or is this behavior specific to my setup?
2. I noticed a +1 in the calculation of effective_block_size (https://github.com/microsoft/Samba/blob/3da69d13348105092424cf870990d27cad545727/pretrain.py#L367). Could you clarify the purpose of adding 1? Specifically, what is meant by "next word" in this context?

**Comment by renll** [comment]
Hi! Thanks for the interests!

1. No, we didn't use a different value. I think there might be something wrong with your setup.

2. We use the objective of next token prediction for pretraining. Specifcally, the target for the cross entropy loss is set as in this line: https://github.com/microsoft/Samba/blob/3da69d13348105092424cf870990d27cad545727/pretrain.py#L238

**Comment by amy-hyunji** [comment]
Thank you for the response!

For the tokenizer, did you use the one from Tinyllama (https://huggingface.co/TinyLlama/TinyLlama_v1.1)?

**Comment by renll** [comment]
Yes, we both use llama2 tokenizer.

**Comment by amy-hyunji** [comment]
Got it! Thanks for the help! :)
